modules/fbs/fbs.module

```diff
@@ -575,3 +620,33 @@ function _fbs_wayf_login($name, $value = NULL) {
 
   return NULL;
 }
+
+/**
+ * Implements hook_ting_pre_execute().
+ *
+ * Filter out blacklisted branches from search.
+ */
+function fbs_ting_pre_execute($request) {
+  // We only want to modify searches.
+  if ($request instanceof TingClientSearchRequest) {
+    $branches_blacklist = array_filter(variable_get('fbs_branches_blacklist', array()));
+    $blacklist_names = array_values(array_intersect_key(fbs_branches(), $branches_blacklist));
+    $blacklist_names = array_map('_fbs_cql_quote_string', $blacklist_names);
+
+    $query = $request->getQuery();
+    if (!empty($branches_blacklist)) {
+      $query .= ' AND (holdingsitem.branch = (* not ' . implode(' not ',
+                $branches_blacklist) . '))';
+      $request->setQuery($query);
+    }
+  }
+}
+
+/**
+ * Quote string for use in CQL.
+ *
+ * Adds " around the string and escapes inline ".
+ */
+function _fbs_cql_quote_string($name) {
+  return '"' . preg_replace('/"/', '//"', $name) . '"';
```

Comment:
When testing it seems like only \\" is working for escaping. So shouldn't this be something like:
`preg_replace('/"/', '\\\\"', $name)`